packages/edit-site/src/components/add-new-pattern/index.js

```diff
@@ -36,21 +36,29 @@ export default function AddNewPattern() {
 	const [ showPatternModal, setShowPatternModal ] = useState( false );
 	const [ showTemplatePartModal, setShowTemplatePartModal ] =
 		useState( false );
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
 	const { createPatternFromFile } = unlock( useDispatch( patternsStore ) );
```

Comment:
Another trick is to use `useRegistry`, call dispatch and unlock in the action callback.